packages/fields-document/src/DocumentEditor/component-blocks/api-shared.ts

```diff
@@ -304,8 +304,8 @@ export type HydratedRelationshipData = {
 
 export type RelationshipData = {
   id: string
-  label: string | undefined
-  data: Record<string, any> | undefined
+  label?: string
+  data?: Record<string, any>
```

Comment:
This is required as a result of `zod` assuming that `| undefined` is equivalent to `?:`, which it isn't.

Reference https://github.com/colinhacks/zod/issues/2464 and https://github.com/colinhacks/zod/issues/2598